spec/ParseAPI.spec.js

```diff
@@ -35,6 +35,36 @@ describe_only_db('mongo')('miscellaneous', () => {
   });
 });
 
+describe_only_db('mongo')('spatial index', () => {
+  beforeAll(async () => {
+    await TestUtils.destroyAllDataPermanently(false);
+    // Create a schema with a spatial index
+    const testSchema = new Parse.Schema('geojson_test');
+    testSchema.addObject('geometry');
+    testSchema.addIndex('geospatial_index', {
+      geometry: '2dsphere',
+    });
+    await testSchema.save();
```

Comment:
I want to make sure we don't introduce a flaky test here. This awaits the schema save, but not the index building in DB.

Maybe you want to look into schema tests to see what do we usually use in these cases, a timeout?

cc @dplewis

Reply:
The test is not flaky for me. The collection starts empty, so index building should take close to no time.
I couldn't find any tests for addIndex on the server codebase. The JS sdk has several, but none of them seem to explicitly wait on index-building.

Reply:
I think there is no way to verify index building via Parse, so the only way would be connecting directly to mongodb and checking the status. Or modify Parse adapter to include that. I believe this test will not be flaky but a timeout (maybe 1 or 2s) could be helpful as well.

Reply:
Ok then let's merge it as it is and see how it goes. I will run the tests a few times here to see whether its flaky already.